app/grants/views.py

```diff
@@ -1400,6 +1400,30 @@ def grant_new_whitelabel(request):
 
     profile = get_profile(request)
 
+    project = None
+    project_id = request.GET.get('project_id', None)
+    if project_id is not None:
+        hackathon_project = HackathonProject.objects.filter(pk=project_id).nocache().first()
+        if hackathon_project is not None:
+            if hackathon_project.grant_obj is None:
+                if request.user.profile in hackathon_project.profiles.all():
```

Comment:
A better (faster) way to do this is `if hackathon_project.profiles.filter(id=request.user.profile.id).exists():` Since we don't actual need the profile object from the queryset.